pkg/shp/reactor/pod_watcher.go

```diff
@@ -73,26 +98,8 @@ func (p *PodWatcher) Start() (*corev1.Pod, error) {
 			if p.skipPodFn != nil && p.skipPodFn(pod) {
 				continue
 			}
-
-			switch event.Type {
-			case watch.Added:
-				if p.onPodAddedFn != nil {
-					if err := p.onPodAddedFn(pod); err != nil {
-						return pod, err
-					}
-				}
-			case watch.Modified:
-				if p.onPodModifiedFn != nil {
-					if err := p.onPodModifiedFn(pod); err != nil {
-						return pod, err
-					}
-				}
-			case watch.Deleted:
-				if p.onPodDeletedFn != nil {
-					if err := p.onPodDeletedFn(pod); err != nil {
-						return pod, err
-					}
-				}
+			if err := p.handleEvent(pod, event); err != nil {
+				return pod, err
 			}
```

Comment:
This looks more like a refactor than part of the Go report card fixes. Was this flagged by the tool?

Reply:
Indeed a minor refactoring due to high cyclomatic-complexity, i.e.:

```
$ goreportcard-cli -v
[...]
	pkg/shp/reactor/pod_watcher.go
		Line 59: warning: cyclomatic complexity 18 of function (*PodWatcher).Start() is high (> 15) (gocyclo)
[...]
```